src/journal-remote/journal-gatewayd.c

```diff
@@ -356,6 +372,82 @@ static int request_parse_range(
         return 0;
 }
 
+static int request_parse_range_time(
+                RequestMeta *m,
+                const char *range) {
+
+        _cleanup_free_ char *until = NULL;
+        char *colon;
+        int r;
+
+        range += 5;
+        range += strspn(range, WHITESPACE);
+
+        colon = strchr(range, ':');
+        if (!colon)
+                return -EINVAL;
+
+        if (colon - range > 0) {
+                _cleanup_free_ char *t = NULL;
+
+                t = strndup(range, colon - range);
+                if (!t)
+                        return -ENOMEM;
+
+                r = parse_sec(t, &m->since);
+                if (r < 0)
+                        return r;
+
+                m->since_set = true;
+        }
+
+        range = colon;
+        colon = strchr(range + 1, ':');
+        if (!colon)
+                until = strdup(range + 1);
+        else {
+                r = request_parse_range_skip_and_n_entries(m, colon);
+                if (r < 0)
+                        return r;
+
+                until = strndup(range + 1, colon - range - 1);
+        }
+        if (!until)
+                return -ENOMEM;
+
+        if (!isempty(until)) {
+                r = parse_sec(until, &m->until);
+                if (r < 0)
+                        return r;
+
+                m->until_set = true;
```

Comment:
Maybe, better to check if `since < until`  here?

Reply:
Right. I got hit by this condition and it returned no value. It's more explicit to return "Invalid header".